nncf/common/tensor_impl_np.py

```diff
@@ -0,0 +1,248 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Dict, Iterator, List, Optional, Tuple, Type, Union
+
+import numpy as np
+from numpy import dtype
+from numpy.ma import MaskedArray
+from numpy.ma.core import MaskedConstant
+
+from nncf import TargetDevice
+from nncf.common.tensor import NNCFTensor
+from nncf.common.tensor import NNCFTensorBackend
+from nncf.common.tensor import TensorDtype
+
+_DTYPE_MAP: Dict[TensorDtype, Any] = {TensorDtype.FLOAT32: np.float32, TensorDtype.INT64: np.int64}
+
+_INV_DTYPE_MAP: Dict[Any, TensorDtype] = {v: k for k, v in _DTYPE_MAP.items()}
+_INV_DTYPE_MAP[dtype("float32")] = TensorDtype.FLOAT32
+
+
+class NPNNCFTensor(NNCFTensor[np.ndarray]):
```

Comment:
This is one such implementation that is currently reused for OV and ONNX, and also can be used for common NNCF-powered code testing with correct functional results.